indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisor.java

```diff
@@ -1103,6 +1104,18 @@ public void stop(boolean stopGracefully)
     }
   }
 
+  @Override
+  public ListenableFuture<Void> stopAsync(boolean stopGracefully)
+  {
+    ListeningExecutorService shutdownExec = MoreExecutors.listeningDecorator(
+        Execs.singleThreaded(StringUtils.encodeForFormat(supervisorId) + "-Shutdown-%d")
```

Comment:
Nit: maybe use a common prefix for all shutdown threads so that it is easier to search for them.
```suggestion
        Execs.singleThreaded("supervisor-shutdown-" + StringUtils.encodeForFormat(supervisorId) + "--%d")
```